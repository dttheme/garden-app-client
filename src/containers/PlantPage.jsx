import React, { Component } from "react";
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchPlant } from '../actions/index.actions';
import Plant from '../components/Plant'

export class PlantPage extends Component {
  componentDidMount() {
      this.props.fetchPlant(this.props.match.params.id);
  }
  render() {
    return (
      <div className="dark-gray bg-dark-green w-100 w-80-m w-60-ns mt7-m mt6-ns ml7-ns ml5-m ba-ns bw3-ns b--green pl6-ns pl4-m pl4 pb4">
          <div>
            <Link to="/garden" className="washed-green no-underline underline-hover hover-light-yellow pr4">{this.props.user.firstName}'s Garden</Link>
            {this.props.currentPlant && <Plant isNotLinking plant={this.props.currentPlant} /> }
          </div>
        }
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentPlant: state.garden.currentPlant,
  user: state.user,
  appState: {
    isFetchingPlant: state.appState.isFetchingPlant,
  },
})

export default connect(mapStateToProps, { fetchPlant })(PlantPage);
