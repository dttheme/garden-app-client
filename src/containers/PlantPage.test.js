import React from 'react';
import { shallow } from 'enzyme';
import { PlantPage } from './PlantPage';

describe('<PlantPage />', () => {

  it('should render page', () => {
    const fakeMatch = {
      params: {
        id: 1
      }
    };
    const fakePlant = {
      id: '123',
      plantName: 'Lettuce'
    };
    const fakeUser= {
      firstName: 'Joe',
    };
    const wrapper = shallow(<PlantPage isNotLinking user={fakeUser} match={fakeMatch} plant={fakePlant} fetchPlant={() => {}}/>);
    expect(wrapper).toMatchSnapshot()
  });

  it('calls fetchPlant on page load', () => {
    const mockFetchPlant = jest.fn();
    const fakeMatch = {
      params: {
        id: 1
      }
    };
    const fakePlant = {
      id: '123',
      plantName: 'Lettuce'
    };
    const fakeUser= {
      firstName: 'Joe',
    };
    const wrapper = shallow(<PlantPage user={fakeUser} match={fakeMatch} plant={fakePlant} fetchPlant={mockFetchPlant} />);
    expect(mockFetchPlant).toHaveBeenCalledTimes(1);
  })
  })
