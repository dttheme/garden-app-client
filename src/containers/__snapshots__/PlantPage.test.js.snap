// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantPage /> should render page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlantPage
    fetchPlant={[Function]}
    isNotLinking={true}
    match={
        Object {
            "params": Object {
              "id": 1,
            },
          }
    }
    plant={
        Object {
            "id": "123",
            "plantName": "Lettuce",
          }
    }
    user={
        Object {
            "firstName": "Joe",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Link
                    className="washed-green no-underline underline-hover hover-light-yellow pr4"
                    replace={false}
                    to="/garden"
          >
                    Joe
                    's Garden
          </Link>
</div>,
        "}",
      ],
      "className": "dark-gray bg-dark-green w-100 w-80-m w-60-ns mt7-m mt6-ns ml7-ns ml5-m ba-ns bw3-ns b--green pl6-ns pl4-m pl4 pb4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="washed-green no-underline underline-hover hover-light-yellow pr4"
              replace={false}
              to="/garden"
>
              Joe
              's Garden
</Link>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Joe",
                "'s Garden",
              ],
              "className": "washed-green no-underline underline-hover hover-light-yellow pr4",
              "replace": false,
              "to": "/garden",
            },
            "ref": null,
            "rendered": Array [
              "Joe",
              "'s Garden",
            ],
            "type": [Function],
          },
          undefined,
        ],
        "type": "div",
      },
      "}",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Link
                        className="washed-green no-underline underline-hover hover-light-yellow pr4"
                        replace={false}
                        to="/garden"
            >
                        Joe
                        's Garden
            </Link>
</div>,
          "}",
        ],
        "className": "dark-gray bg-dark-green w-100 w-80-m w-60-ns mt7-m mt6-ns ml7-ns ml5-m ba-ns bw3-ns b--green pl6-ns pl4-m pl4 pb4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="washed-green no-underline underline-hover hover-light-yellow pr4"
                replace={false}
                to="/garden"
>
                Joe
                's Garden
</Link>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Joe",
                  "'s Garden",
                ],
                "className": "washed-green no-underline underline-hover hover-light-yellow pr4",
                "replace": false,
                "to": "/garden",
              },
              "ref": null,
              "rendered": Array [
                "Joe",
                "'s Garden",
              ],
              "type": [Function],
            },
            undefined,
          ],
          "type": "div",
        },
        "}",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
